### PR TITLE
Feature/us 020 put operation update a user

### PR DESCRIPTION
**Pull request by ATdllerenas**
Implementation of the update operation for users.

### CODE REVIEW

**Review by luisgalindoluna — CHANGES_REQUESTED** [review]
Please solve conflicts first

File: src/main/java/com/agilethought/atsceapi/dto/UpdateUserResponse.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@Getter
+@Setter
+public class UpdateUserResponse {
+	//The Response will be a 200 HttpStatus.OK with the follow Response Body
+	private String id;
+	private Integer type;
+	private String firstName;
+	private String lastName;
+	private String email;
+	private String password;
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/dto/UpdateUserResponse.java]: I think password must not be included as por of response. What do you think @EmiliaGuillen-dev?

File: src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java [hunk 2]
```diff
@@ -1,81 +1,114 @@
...
+			
+			User savedUsers = userRepository.save(user);
+			
+			response.setId(user.getId());
+			response.setType(user.getType());
+			response.setFirstName(user.getFirstName());
+			response.setLastName(user.getLastName());
+			response.setEmail(user.getEmail());
+			response.setPassword(user.getPassword());
+			response.setStatus(user.getStatus());
```
Comment (luisgalindoluna) [inline src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java]: Your mapper already does this

**Review by luisgalindoluna — CHANGES_REQUESTED** [review]
Overall looks good, just use your mapper for your updateUser method

File: src/main/java/com/agilethought/atsceapi/dto/UpdateUserResponse.java (hunk 1 above)
Comment (EmiliaGuillen-dev) [reply src/main/java/com/agilethought/atsceapi/dto/UpdateUserResponse.java]: I totally agree

Comment (ATdllerenas) [reply src/main/java/com/agilethought/atsceapi/dto/UpdateUserResponse.java]: So is the only change missing? I will make de change.

**Review by luisgalindoluna — APPROVED** [review]
Well done

File: src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java [hunk 3]
```diff
@@ -1,81 +1,106 @@
...
+			user.setEmail(request.getEmail());
+			user.setPassword(request.getPassword());
+			user.setStatus(request.getStatus()); 			
+			
+			User savedUsers = userRepository.save(user);
+			
+			return orikaMapperFacade.map(savedUsers, UpdateUserResponse.class);
+		
+		}else {
+			throw new NotFoundException("User Not Found with id: " + Id);
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java:102]: Just as recommendation, it is a good practice have a constant class, where we define constant messegas such as `private static final String USER_NOT_FOUND =  "User Not Found with id: %s"`.

**Review by ivanflores15 — APPROVED** [review]
looks good

**Review by EmiliaGuillen-dev — APPROVED** [review]
looks good

**Pull request merged**